src/l500/l500-color.cpp

```diff
@@ -184,33 +184,40 @@ namespace librealsense
 
         auto num_of_res = intrinsic->resolution.num_of_resolutions;
 
-        for( auto i = 0; i < num_of_res; i++ )
+        if (num_of_res <= MAX_NUM_OF_RGB_RESOLUTIONS)
         {
-            auto model = intrinsic->resolution.intrinsic_resolution[i];
-            if( model.height == profile.height && model.width == profile.width )
+            for (auto i = 0; i < num_of_res; i++)
             {
-                rs2_intrinsics intrinsics;
-                intrinsics.width = model.width;
-                intrinsics.height = model.height;
-                intrinsics.fx = model.ipm.focal_length.x;
-                intrinsics.fy = model.ipm.focal_length.y;
-                intrinsics.ppx = model.ipm.principal_point.x;
-                intrinsics.ppy = model.ipm.principal_point.y;
-
-                if( model.distort.radial_k1 || model.distort.radial_k2 || model.distort.tangential_p1 || model.distort.tangential_p2 || model.distort.radial_k3 )
+                auto model = intrinsic->resolution.intrinsic_resolution[i];
+                if (model.height == profile.height && model.width == profile.width)
                 {
-                    intrinsics.coeffs[0] = model.distort.radial_k1;
-                    intrinsics.coeffs[1] = model.distort.radial_k2;
-                    intrinsics.coeffs[2] = model.distort.tangential_p1;
-                    intrinsics.coeffs[3] = model.distort.tangential_p2;
-                    intrinsics.coeffs[4] = model.distort.radial_k3;
+                    rs2_intrinsics intrinsics;
+                    intrinsics.width = model.width;
+                    intrinsics.height = model.height;
+                    intrinsics.fx = model.ipm.focal_length.x;
+                    intrinsics.fy = model.ipm.focal_length.y;
+                    intrinsics.ppx = model.ipm.principal_point.x;
+                    intrinsics.ppy = model.ipm.principal_point.y;
+
+                    if (model.distort.radial_k1 || model.distort.radial_k2 || model.distort.tangential_p1 || model.distort.tangential_p2 || model.distort.radial_k3)
+                    {
+                        intrinsics.coeffs[0] = model.distort.radial_k1;
+                        intrinsics.coeffs[1] = model.distort.radial_k2;
+                        intrinsics.coeffs[2] = model.distort.tangential_p1;
+                        intrinsics.coeffs[3] = model.distort.tangential_p2;
+                        intrinsics.coeffs[4] = model.distort.radial_k3;
 
-                    intrinsics.model = RS2_DISTORTION_INVERSE_BROWN_CONRADY;
-                }
+                        intrinsics.model = RS2_DISTORTION_INVERSE_BROWN_CONRADY;
```

Comment:
It's not clear to me what the model is if we don't get into this if()... can you look into that and fix it, please?

Reply:
I'm not sure what should be the default.
 RS2_DISTORTION_BROWN_CONRADY? RS2_DISTORTION_NONE?

for ds5 they set it to RS2_DISTORTION_BROWN_CONRADY on all cases except fisheye sensor.
what do you think?

Reply:
BROWN

Reply:
(that's the default regardless, but I'll ask the algo team)

Reply:
From Aviad's answer:
Remove the code that checks the zeroes in the distortion and always assign inverse brown, with a TODO explaining that it's wrong and should be changed to regular brown.

Reply:
Will be updated.